fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -77,6 +79,21 @@
             SYSTEM_CONFIG.getPackageVariableName("lucene_search_timeout_ms"), 600000
     );
 
+    public static final LRUQueryCache LUCENE_QUERY_CACHE;
+    static {
+        // Lucene 5.3.0 disables query caching.  Create cache for use here.
+        // min of 32MB or 5% of the heap size.  Same default as Lucene 5.4.0.
+        final long maxRamBytesUsed = SYSTEM_CONFIG.getLongProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_size_bytes"),
+            Math.min(1L << 25, Runtime.getRuntime().maxMemory() / 20));
+
+        // default max cached queries based on memory used
+        final int maxCachedQueries = SYSTEM_CONFIG.getIntProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_queries"),
+            (int) Math.max(1000L, Math.min(1000000L, maxRamBytesUsed / 1000)));
+        LUCENE_QUERY_CACHE = new LRUQueryCache(maxCachedQueries, maxRamBytesUsed);
+    }
+
```

Comment:
Let's leave it outstanding for a little bit longer. It would be good to confirm that the most recent version of Fili (the version with the updated Lucene dependency) fixes you guys. If we can confirm that, I'm cool closing this PR.

If you don't want / can't take the most recent dev version of Fili, I _think_ you should be able to bump your local Lucene dependency, since the Fili PR to bump it didn't require any code changes.